modules/heat_conduction/src/actions/ThermalContactBCsAction.C

```diff
@@ -49,7 +49,8 @@ ThermalContactBCsAction::act()
   InputParameters params = _factory.getValidParams(getParam<std::string>("type"));
 
   // Extract global params
-  _app.parser().extractParams(_name, params);
+  if (isParamValid("parser_syntax"))
+    _app.parser().extractParams(getParam<std::string>("parser_syntax"), params);
```

Comment:
I can add error, but is it an error? This implies that these actions **must** be added by the parser and not by a user via a meta action, which may be true.

This is where your smart parameter lookup stuff will eliminate the need for any of this.